File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        maxsum = nums[0]
        curr =0
        i=0
        while i<len(nums):
            if curr < 0:
                curr =0
            curr += nums[i]
            maxsum = max(curr,maxsum)
            i+=1
        return maxsum        

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] nums) {
        
        int maxarea =0;
        int start =0;
        int end =nums.length-1;
        while(start<end){
            maxarea = Math.max(Math.min(nums[start],nums[end]) * (end - start), maxarea);
             if (nums[start] < nums[end]) {
                start++;
            } else {
                end--;
            }
        }
        return maxarea;
    }
}
